internal/connectors/engine/engine.go

```diff
@@ -716,6 +716,20 @@ func (e *engine) CreatePool(ctx context.Context, pool models.Pool) error {
 	ctx, span := otel.Tracer().Start(ctx, "engine.CreatePool")
 	defer span.End()
 
+	for _, poolAccount := range pool.PoolAccounts {
+		exists, err := e.storage.AccountsExists(ctx, poolAccount.AccountID)
```

Comment:
This probably needs to be in the same transaction as PoolsUpsert or else there could still be inconsistencies.

Reply:
I can do it directly in the storage yes